kitsune/search/v2/base.py

```diff
@@ -274,10 +280,25 @@ def get_highlight_fields(self):
         pass
 
     @abstractmethod
-    def get_filter(self):
+    def get_filter(self, **kwargs):
         """A query which filters for all documents to be searched over."""
         pass
 
+    def build_query(self, **kwargs):
+        """Build a query to search over a specific set of documents."""
+        query = kwargs.get("query", "")
+        operator = kwargs.get("default_operator", "AND")
+        return DSLQ(
+            "simple_query_string",
+            query=query,
+            default_operator=operator,
+            fields=self.get_fields(),
+            # everything apart from WHITESPACE as that interferes with char mappings
+            # and synonyms with whitespace in them by breaking up the phrase into tokens,
+            # before they have a chance to go through the filter:
```

Comment:
I'd remove the comment here, because we have it above the constant - we could also remove the constant, since the simple_query_string is now DRY